### ISSUE TITLE
bazel-dev crashes if `JAVA_HOME` points to JDK 8

### ISSUE THREAD

**Issue by fmeum** [bazelbuild/bazel#16850]
### Description of the bug:

Since https://github.com/bazelbuild/bazel/commit/94355b1b1c4f7ef923457b8b2a070e5c6528240a, the binary built with `//src:bazel-dev`, but not the one built with `//src:bazel`, crashes when `JAVA_HOME` points to a JDK 8:

```
Starting local Bazel server and connecting to it...
Server crashed during startup. Now printing /home/fhenneke/.cache/bazel/_bazel_fhenneke/796b3b05d259febdf80296a632dc2e84/server/jvm.out
Unrecognized option: --add-opens=java.base/java.lang=ALL-UNNAMED
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

1. `bazel build //src:bazel-dev`
2. `JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64/ bazel-bin/src/bazel-dev build //src:bazel-dev`

### Which operating system are you running Bazel on?

Linux

**Comment by fmeum** [comment]
@cushon I wanted to report this behavior as I didn't expect this to happen when I made the `--add-opens` flag unconditional. Do you see a way to fix this without losing the benefits of https://github.com/bazelbuild/bazel/commit/94355b1b1c4f7ef923457b8b2a070e5c6528240a for non-Bazel devs?

**Comment by cushon** [comment]
> Do you see a way to fix this without losing the benefits of [94355b1](https://github.com/bazelbuild/bazel/commit/94355b1b1c4f7ef923457b8b2a070e5c6528240a) for non-Bazel devs?

I don't see great ways for the Bazel launcher to detect what JDK version is being used. As long as `//src:bazel` has dropped support for JDK 8, it seems OK to me for `//src:bazel-dev` to match that and also require a newer version.

It would be nice if there was a good way to report a nicer error message, but I'm also not sure that needs to be high priority.
